### ISSUE TITLE
Incorrect order of psuedoelements and classes

### ISSUE THREAD

**Issue by matthughes** [japgolly/scalacss#36]
Psuedo elements (before/after) must come last after psueudo classes (:hover/:visited) according to spec (http://www.w3.org/TR/selectors/#selector-syntax).

Currently, no matter which order one declares elements/classes, elements appear before classes:

```
  val foo = style("foo")(
    &.hover.before(color(blue))
  )
```
generates:
```
.foo::before:hover { color: blue }
```

Should be:

```
.foo:hover::before { color: blue }
```

**Comment by matthughes** [comment]
I'm going to try a PR for this.

**Comment by japgolly** [comment]
Merged #40. Big thank you for the PR, @matthughes !
